### DISCUSSION TITLE
Is it possible to remove shadow drop effect from window or control how rounded corners of window are without using system settings?

### DISCUSSION THREAD

**Discussion by BreyerW** [AvaloniaUI/Avalonia#12963] [Q&A]
im trying to use semi-transparent window as overlay (can be partially outside of other windows as such i think i must use window for that purpose) and ideally i need to at least remove shadow effect from specific window. Being able to control how rounded corners are is a bonus but not a must

Is this possible in Avalonia? Standard `Window` doesnt seem to have anything related to these parameters

EDIT: `CornerRadiusProperty` doesnt seem to work on Win 11 at least

**Comment by thevortexcloud** [comment]
Generally speaking, window decorations are supplied by the operating system. You do not have any real control over it. You can however hide the operating system decorations and creating your own. But this may make your app feel less than native on a lot of platforms and requires more work.

**Reply by BreyerW** [reply]
I dont need decorations. Indeed since we are talking about window as overlay i want as little decorations as possible and already removed most. Only shadow effect and rounded corners are left.

**Reply by thevortexcloud** [reply]
If you don't want the frame at all you can just completely disable it in Avalonia by doing:

```xaml
  SystemDecorations="None"
        ExtendClientAreaChromeHints="NoChrome"
        ExtendClientAreaToDecorationsHint="True"
```
As options on your window. Beyond that you probably are going to have to resort to native APIs/making your own custom frame in Avalonia.

**Reply by BreyerW** [reply]
As i said i already did that. As for system api i would like to avoid that if possible

**Reply by timunie** [reply]
See also what headaches it is in WPF or WinUI to get rid of the round corners etc: https://learn.microsoft.com/en-us/windows/apps/desktop/modernize/apply-rounded-corners?WT.mc_id=WD-MVP-5001077

**Comment by xmaxrayx** [comment]
yeah was wired why it was easy do it on AHK bc it was Normal old window style 
avalonia and other like WPF use layerdEX inside new window system which seems disable shadow being ignored with DWM api,

maybe in future windows add it like their backdrop API lol

for me i just draw boreder
```
        Background="Transparent"
        SystemDecorations="None"
        ExtendClientAreaToDecorationsHint="False"
        ExtendClientAreaChromeHints="NoChrome"
        ExtendClientAreaTitleBarHeightHint="0"
        CanResize="False"
        ```

**Reply by xmaxrayx** [reply]
ok i tired add corner window from windows DWM api and I saw they add shadow with it sadly
so you need make the border internally.

maybe it will change in future,
